lib/pkcs11h-threading.c

```diff
@@ -441,6 +446,11 @@ _pkcs11h_threading_condWait (
 		timeout.tv_sec = now.tv_sec + milli/1000;
 		timeout.tv_nsec = now.tv_usec*1000 + milli%1000;
 
+		if (pthread_mutex_trylock (&cond->mut)) {
+			rv = CKR_FUNCTION_FAILED;
+			goto cleanup;
+		}
+		unlock_mutex = 1;
```

Comment:
this should wait not fail if other holds the lock, can it be done?

Reply:
We could spin in a loop calling `pthread_mutex_trylock` until times passes out. But this is basically a spinlock.
Not sure if this is a good idea.

Reply:
as far as I remember this was my initial use of cond as mutex cannot wait... yes, spin loop is bad, there must be a different solution, otherwise the `pthread_cond_timedwait` has little sense.

Reply:
If the mutex is only used for waiting on the condition variable, the amount of time it is locked should be really short. Maybe spinlock here is actually OK.

Anyways, this change doesn't make the code any worse than it was.